src/cursor/find_cursor.ts

```diff
@@ -68,31 +68,29 @@ export class FindCursor<TSchema = any> extends AbstractCursor<TSchema> {
   }
 
   /** @internal */
-  _initialize(session: ClientSession, callback: Callback<ExecutionResult>): void {
+  async _initialize(session: ClientSession): Promise<ExecutionResult> {
     const findOperation = new FindOperation(undefined, this.namespace, this[kFilter], {
       ...this[kBuiltOptions], // NOTE: order matters here, we may need to refine this
       ...this.cursorOptions,
       session
     });
 
-    executeOperation(this.client, findOperation, (err, response) => {
-      if (err || response == null) return callback(err);
+    const response = await executeOperation(this.client, findOperation);
 
-      // TODO: We only need this for legacy queries that do not support `limit`, maybe
-      //       the value should only be saved in those cases.
-      if (response.cursor) {
-        this[kNumReturned] = response.cursor.firstBatch.length;
-      } else {
-        this[kNumReturned] = response.documents ? response.documents.length : 0;
-      }
+    // TODO: We only need this for legacy queries that do not support `limit`, maybe
+    //       the value should only be saved in those cases.
+    if (response.cursor) {
+      this[kNumReturned] = response.cursor.firstBatch.length;
+    } else {
+      this[kNumReturned] = response.documents ? response.documents.length : 0;
+    }
 
-      // TODO: NODE-2882
-      callback(undefined, { server: findOperation.server, session, response });
-    });
+    // TODO: NODE-2882
+    return { server: findOperation.server, session, response };
   }
 
   /** @internal */
-  override _getMore(batchSize: number, callback: Callback<Document>): void {
+  override async getMore(batchSize: number): Promise<Document | null> {
     // NOTE: this is to support client provided limits in pre-command servers
```

Comment:
This comment concerns me. I imagine this logic is only used by the mock server. Maybe this isn't the PR to address it but we've edited this code twice despite allegedly being outside our supported server matrix. Any thoughts on best path forward?

Reply:
I removed the pre-3.6 logic but left the manual limit enforcing code because there is a dedicated drivers ticket to address it.